rnd/market/schema.prisma

```diff
@@ -11,6 +11,22 @@ generator client {
   partial_type_generator = "market/utils/partial_types.py"
 }
 
+enum UserRole {
+  USER
+  ADMIN
+}
+
+model User {
+  id    String  @id // This should match the Supabase user ID
+  role  UserRole    @default(USER)
```

Comment:
Can we pull this from Supabase? Maybe we are idk?

Reply:
Roles should be coming from Supabase. In server I haven't set up roles yet to sync with Supabase. Does marketplace need roles?

Reply:
Yes, we need to approve new posts and promote posts to featured but only admins can do that

Reply:
Admins = autogpt employees right? or is there potential for external users too?

Reply:
Currently employees. Not sure long term but maybe?

Reply:
Probably not at admin level. Would likely be dedicated role

Reply:
okay cool. The plan for roles was this: 

Table in supabase that contains roles as well (right now we dont have roles) 
This role info would then come from supabase itself, via the jwt token (in claims field) 
And same as how we keep sync all other user info (that's not private, e.g. names), we keep the role too

Reply:
Yeah this is the plan we discussed on Discord. This PR is in draft as it will be modified as we are ready to integrate fully with supabase